core/src/main/java/com/linecorp/armeria/common/metric/TimedExecutor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linecorp.armeria.common.metric;
+
+import java.util.concurrent.Executor;
+
+import com.linecorp.armeria.common.util.UnstableApi;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+
+/**
+ * An {@link Executor} that is timed.
+ */
+@UnstableApi
+public class TimedExecutor implements Executor {
```

Comment:
How about hiding everything but `ExecutorServiceMetrics` to minimize the amount of forked public API?

Reply:
As you mentioned https://github.com/line/armeria/pull/2612#discussion_r395122626, if `ExecutorServiceMetrics` is moved to `common`, we need to move to `TimedExecutor` to `common` to remove `public` modifier. 
If it's fine, I can make it package private. 😀